### DISCUSSION TITLE
Qual melhor framework para Material Design?

### DISCUSSION THREAD

**Discussion by thaisnoboss** [frontendbr/forum#418] [Perguntas e Respostas]
Olá a todos,

Estamos realizando um projeto no qual precisamos decidir quais frameworks iremos utilizar. Durante nossas pesquisas descobrimos o Material Design do Google que seria útil para o desenvolvimento da interface administrativa, porém encontramos poucas informações e comparativos sobre qual framework baseado no Material Design é o mais utilizado, ou o melhor. 

Gostaria de saber se alguém já utilizou o Materialize e ou Material Design Lite para que possamos ter opiniões comparativas entre eles. 

Seria ótimo também, sugestões de frameworks atuais e ferramentas que estão sendo utilizados atualmente.

Nosso maior receio é que o Material Design Lite (que já foi descontinuado pelo google) ou o próprio Materialize estejam defasados. 

Extra: Alguém já leu sobre, ou já tentou usar o Material Components do google? 

Desde já agradeço!

**Comment by joao-alberto** [comment]
Antes de tudo, qual será o stack de tecnologias que vocês planejam usar?

Por exemplo, vocês planejam usar angular no projeto? Caso sim, eu recomendo o [Angular Material](https://material.angularjs.org/latest/)
Se usar React, eu recomendo [Material UI](https://github.com/callemall/material-ui)

Mas se não for nenhum dos dois, eu ainda recomento o [Material Design Lite](https://getmdl.io/).

> No further development is taking place in MDL by the core team, but we are happy to review PRs, fix critical bugs and push out new releases. No breaking changes will be accepted

Ele não esta sendo descontinuado, continuará recendo atualizações e lançando releases, entretanto apenas da comunidade, e não mais do time do core do Google.

Espero ter ajudado. :v:

**Comment by thaisnoboss** [comment]
Olá!

Ajudou sim, de fato já tínhamos o conhecimento desse suporte limitado, porém como será um produto que terá uma vida longa por assim dizer, nos preocupamos com o suporte a longo prazo e quanto seria o retrabalho ocasionado futuramente até mesmo para migrar, por exemplo, para o Material Components. 

Estamos pensando sim no uso do React, mas como a equipe não possui conhecimento na ferramenta tememos o tamanho da curva de aprendizagem, mas é uma alternativa que estamos avaliando.
 
Mas só por curiosidade, você tem alguma opinião específica sobre o Materialize como framework baseado em Material Design? Pois cheguei a investigar e realizar alguns testes e considerei ela de certa forma limitada, apesar de em alguns fóruns e votações ela se sobressair em relação ao Material Design Lite.

Abraços!

**Comment by joao-alberto** [comment]
Eu acho que ambos são bem equivalentes, pelo menos no que tange aos elementos descritos no manual do Material Design.

A maior diferença entre eles é que o Materialize tem toda uma seção de elementos que utilizam JavaScript para funcionar (Carousel, Collapsible, Parallax e outros) e que não estão disponíveis no Material Design Lite. A questão é, você utilizará esses elementos?

Sobre o desenvolvimento com ambas, elas tem um nível de dificuldade igual (baixo). É apenas ler a documentação e utilizar as classes.

**Comment by danvitoriano** [comment]
E ae galera, me gerou uma dúvida aqui. Busquei na net mas não achei.
Aonde vc leu @thaisnoboss que o Material Design Lite foi descontinuado?

> Material Design Lite (que já foi descontinuado pelo google)

**Comment by joao-alberto** [comment]
@danvitoriano Foi [daqui](https://github.com/google/material-design-lite#limited-support)

Mas como comentei ali [no topo](https://github.com/frontendbr/forum/issues/418#issuecomment-275193804), não será descontinuado.

**Comment by danvitoriano** [comment]
que confuso isso @psicotropicos 
eles falam q aceitam PRS mas tem um zilhão de issues e PRs desde novembro #medo

**Comment by thaisnoboss** [comment]
@danvitoriano foi um erro de colocação, na verdade ele está apenas com suporte limitado. 

Também tem mais detalhes sobre a migração do Lite pro Components em uma relase do Lite [aqui](https://github.com/google/material-design-lite/releases)  

Olhe no tópico "MDC-Web and plans for the future"

**Comment by GustavoCostaW** [comment]
Cara se você estiver trabalhando com Angular vai de https://material.angular.io está em beta e todo mundo tá trabalhando nele, em breve vai aparecer mais components, a versão do material para o AngularJS só está recebendo bug fix
